Add tracking when opening or searching in the Command Palette

Aljullu opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/41837.

This PR adds tracking when the user opens the Command Palette and makes a search in it.

cc @pmcpinto

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. To test tracking, open the browser console (<kbd>F12</kbd>) and run `localStorage.setItem( 'debug', 'wc-admin:*' );`.
1. Go to the post editor (Posts > Add New).
2. Press <kbd>Ctrl</kbd>+<kbd>K</kbd> to open the Command Palette.
3. Verify an event is triggered when the Command Palette was opened (in the browser console you will see `wc-admin:tracks recordevent wcadmin_woocommerce_command_palette_open`).
4. Type a search term in the Command Palette. Ie, `products`.
5. An event should be triggered with the search term (in the browser console you will see `wc-admin:tracks recordevent wcadmin_woocommerce_command_palette_search 
{ value: "products", origin: "post-editor" }`).

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment

Add tracking when opening or searching in the Command Palette

</details>

Aljullu committed:
Add tracking when opening or searching in the Command Palette

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

dinhtungdu reviewed (approved):
Code looks and tests good in my test so I'm pre-approving! Thank you for working on this.

I have a question on the palette open event. Do we record where (which screen/page) the palette is opened? From the debug value I didn't see any data related to the current screen. I also see a property named `shouldRecord` set to `false`, which is a bit confusion and makes me wonder if the event was recorded or not.

![image](https://github.com/woocommerce/woocommerce/assets/5423135/9a56c34f-d0ec-4773-a51e-6290256f62a9)

Aljullu committed:
Add 'origin' property to all Command Palette events

Aljullu commented:
> I have a question on the palette open event. Do we record where (which screen/page) the palette is opened?

Good idea! I added an `origin` property to all events which contains the origin editor, which can be: `post-editor`, `page-editor`, `wp_template-editor` or `wp_template_part-editor`. I wasn't sure if we should unify those last two into `site-editor`, but I felt it made mroe sense to always include the `post_type` being edited. This way, if in the future the command palette is displayed in the Product editor, we can have `product-editor` as an `origin`.

Btw, I added this property to all events, not only the open one. That means also when submitting one command or when searching. I think it might be useful in those cases as well, but cc'ing @pmcpinto.

> I also see a property named `shouldRecord` set to `false`, which is a bit confusion and makes me wonder if the event was recorded or not.

I think that's expected if you are in development mode or don't have tracking enabled:

https://github.com/woocommerce/woocommerce/blob/be79b9864e8b5fe6c6acd9d392cc149464e41fe6/packages/js/tracks/src/index.ts#L38-L42

dinhtungdu reviewed (approved):
The update looks good to me! Tested and I can confirm it's working, the `origin` data is tracked correctly.
